internal/services/cdn/cdn_frontdoor_route_resource.go

```diff
@@ -268,13 +277,19 @@ func resourceCdnFrontDoorRouteCreate(d *pluginsdk.ResourceData, meta interface{}
 		originGroup = expandResourceReference(id.ID())
 	}
 
-	normalizedRuleSets, err := expandRuleSetIds(ruleSetIdsRaw)
+	if !features.FourPointOhBeta() {
```

Comment:
Could we consolidate this down to make cleanup easier? I think we could cut down on these multiple checks for FourPointOhBeta and instead we could package it all into one check after instantiating `props` like so:

```
props := cdn.Route{
...
}

if !features.FourPointOhBeta() {
  props.RuleSets =  expandRuleSetReferenceArray(d.Get("cdn_frontdoor_rule_set_ids").(*pluginsdk.Set).List()),
}
```